src/lib/components/chat/ChatMessage.svelte

```diff
@@ -202,13 +252,103 @@
 		<div
 			class="relative min-h-[calc(2rem+theme(spacing[3.5])*2)] min-w-[60px] break-words rounded-2xl border border-gray-100 bg-gradient-to-br from-gray-50 px-5 py-3.5 text-gray-600 prose-pre:my-2 dark:border-gray-800 dark:from-gray-800/40 dark:text-gray-300"
 		>
+			{#if message.files && message.files.length > 0}
+				<div class="grid w-fit grid-cols-2 gap-5">
+					{#each message.files as file}
+						<!-- handle the case where this is a hash that points to an image in the db, hash is always 64 char long -->
+						<button on:click={() => (modalImageToShow = file)}>
+							{#if file.type === "hash"}
+								<img
+									src={urlNotTrailing + "/output/" + file.value}
+									alt="output from assistant"
+									class="my-2 aspect-auto max-h-48 cursor-pointer rounded-lg shadow-lg"
+								/>
+							{:else}
+								<!-- handle the case where this is a base64 encoded image -->
+								<img
+									src={`data:${file.mime};base64,${file.value}`}
+									alt="output from assistant"
+									class="my-2 aspect-auto max-h-48 cursor-pointer rounded-lg shadow-lg"
+								/>
+							{/if}
+						</button>
+					{/each}
+				</div>
+			{/if}
 			{#if searchUpdates && searchUpdates.length > 0}
 				<OpenWebSearchResults
 					classNames={tokens.length ? "mb-3.5" : ""}
 					webSearchMessages={searchUpdates}
 				/>
 			{/if}
 
+			{#if tools}
+				{#each Object.values(tools) as tool}
+					{#if tool.length > 0}
+						{@const toolName = tool.find(isMessageToolCallUpdate)?.call.name}
+						{@const toolDone = tool.some(isMessageToolResultUpdate)}
+						{#if toolName && toolName !== "websearch"}
+							<details
+								class="group/tool my-2.5 w-fit cursor-pointer rounded-lg border border-gray-200 bg-white pl-1 pr-2.5 text-sm shadow-sm transition-all open:mb-3
+								open:border-purple-500/10 open:bg-purple-600/5 open:shadow-sm dark:border-gray-800 dark:bg-gray-900 open:dark:border-purple-800/40 open:dark:bg-purple-800/10"
+							>
+								<summary
+									class="flex select-none list-none items-center gap-1.5 py-1 group-open/tool:text-purple-700 group-open/tool:dark:text-purple-300"
+								>
+									<div
+										class="relative grid size-[22px] place-items-center rounded bg-purple-600/10 dark:bg-purple-600/20"
+									>
+										<svg
```

Comment:
should we move this svg to the icons folder? Afterwards you can use `IconXyz class={!isToolDone ? "opacity-100" : "opacity-0"}`

Reply:
I think it would be good idea to move it into separate icon file. Wdyt @Saghen ?